File: scripts/bump-packages.py
#!/usr/bin/env python3
"""Bulk-update package.json files across the workspace."""

import json
import os
from pathlib import Path

REPO = Path("/home/vscode/repo")

# Common alignment bumps applied to ALL package.json files
COMMON_BUMPS = {
    ("devDependencies", "@cloudflare/vite-plugin", "1.30.1"): "1.33.2",
    ("devDependencies", "@cloudflare/vite-plugin", "1.31.0"): "1.33.2",
    ("devDependencies", "@cloudflare/workers-types", "4.20260331.1"): "4.20260426.1",
    ("devDependencies", "wrangler", "4.77.0"): "4.85.0",
    ("devDependencies", "wrangler", "4.79.0"): "4.85.0",
    ("devDependencies", "@types/node", "24.10.4"): "~25.6.0",
    ("devDependencies", "@types/node", "~24.12.0"): "~25.6.0",
    ("devDependencies", "typescript", "5.9.3"): "6.0.3",
    ("devDependencies", "typescript", "6.0.2"): "6.0.3",
    ("dependencies", "@cloudflare/workers-types", "4.20260331.1"): "4.20260426.1",
    ("dependencies", "@cloudflare/workers-types", "4.20260405.1"): "4.20260426.1",
    ("dependencies", "@types/node", "24.10.4"): "~25.6.0",
    ("dependencies", "@types/node", "~24.12.0"): "~25.6.0",
    ("dependencies", "typescript", "5.9.3"): "6.0.3",
    ("dependencies", "typescript", "6.0.2"): "6.0.3",
}

# Per-package bumps for specific files
# Format: (relative_path, dep_type, package_name, old_version): new_version
SPECIFIC_BUMPS = {
    # docs
    ("docs/package.json", "dependencies", "@base-ui/react", "^1.2.0"): "^1.4.1",
    ("docs/package.json", "dependencies", "fumadocs-core", "^16.7.6"): "^16.8.5",
    ("docs/package.json", "dependencies", "fumadocs-mdx", "^14.2.7"): "^14.3.2",
    ("docs/package.json", "dependencies", "fumadocs-ui", "npm:@fumadocs/base-ui@latest"): "npm:@fumadocs/base-ui@latest",
    ("docs/package.json", "dependencies", "tailwindcss", "^4.1.13"): "^4.2.4",
    ("docs/package.json", "devDependencies", "@fumadocs/base-ui", "^16.7.4"): "^16.8.5",
    ("docs/package.json", "devDependencies", "@tailwindcss/vite", "^4.1.6"): "^4.2.4",
    ("docs/package.json", "devDependencies", "oxfmt", "^0.43.0"): "^0.47.0",
    ("docs/package.json", "devDependencies", "oxlint", "^1.58.0"): "^1.62.0",
    # playground/mantine
    ("playground/mantine/package.json", "dependencies", "@mantine/core", "^9.0.0"): "^9.1.1",
    ("playground/mantine/package.json", "dependencies", "@mantine/hooks", "^9.0.0"): "^9.1.1",
    ("playground/mantine/package.json", "devDependencies", "postcss", "^8.5.8"): "^8.5.12",
    # playground/shadcn
    ("playground/shadcn/package.json", "dependencies", "lucide-react", "^1.7.0"): "^1.11.0",
    ("playground/shadcn/package.json", "dependencies", "react-hook-form", "^7.72.0"): "^7.74.0",
    ("playground/shadcn/package.json", "dependencies", "react-resizable-panels", "^4.7.2"): "^4.10.0",
    ("playground/shadcn/package.json", "dependencies", "tailwindcss", "^4.2.1"): "^4.2.4",
    ("playground/shadcn/package.json", "devDependencies", "@tailwindcss/vite", "^4.2.1"): "^4.2.4",
    ("playground/shadcn/package.json", "devDependencies", "shadcn", "^4.1.2"): "^4.5.0",
    # playground/resend
    ("playground/resend/package.json", "dependencies", "@react-email/components", "^1.0.8"): "^1.0.12",
    ("playground/resend/package.json", "dependencies", "@react-email/render", "^2.0.4"): "^2.0.7",
    ("playground/resend/package.json", "dependencies", "resend", "^6.10.0"): "^6.12.2",
    # playground/storybook
    ("playground/storybook/package.json", "devDependencies", "@storybook/addon-a11y", "^10.3.3"): "^10.3.5",
    ("playground/storybook/package.json", "devDependencies", "@storybook/addon-docs", "^10.3.3"): "^10.3.5",
    ("playground/storybook/package.json", "devDependencies", "@storybook/addon-links", "^10.3.3"): "^10.3.5",
    ("playground/storybook/package.json", "devDependencies", "@storybook/addon-vitest", "^10.3.3"): "^10.3.5",
    ("playground/storybook/package.json", "devDependencies", "@storybook/react", "^10.3.3"): "^10.3.5",
    ("playground/storybook/package.json", "devDependencies", "@storybook/react-vite", "^10.3.3"): "^10.3.5",
    ("playground/storybook/package.json", "devDependencies", "storybook", "^10.3.3"): "^10.3.5",
    # playground/content-collections
    ("playground/content-collections/package.json", "dependencies", "@content-collections/core", "^0.14.3"): "^0.15.0",
    ("playground/content-collections/package.json", "dependencies", "@content-collections/vite", "^0.2.9"): "^0.3.0",
    # playground/community/todo-serverquery-and-actions
    ("playground/community/todo-serverquery-and-actions/package.json", "dependencies", "tailwindcss", "^4.2.1"): "^4.2.4",
    ("playground/community/todo-serverquery-and-actions/package.json", "devDependencies", "@tailwindcss/vite", "^4.2.1"): "^4.2.4",
    # playground/database-do
    ("playground/database-do/package.json", "dependencies", "kysely", "^0.28.15"): "^0.28.16",
    # community
    ("community/package.json", "devDependencies", "@cloudflare/workers-types", "~4.20260331.1"): "~4.20260426.1",
    ("community/package.json", "devDependencies", "vitest", "~4.1.2"): "~4.1.5",
    # community/playground/ark-ui-showcase
    ("community/playground/ark-ui-showcase/package.json", "dependencies", "@ark-ui/react", "^5.35.0"): "^5.36.2",
    ("community/playground/ark-ui-showcase/package.json", "dependencies", "lucide-react", "^1.7.0"): "^1.11.0",
    # community/playground/chakra-ui-showcase
    ("community/playground/chakra-ui-showcase/package.json", "dependencies", "@chakra-ui/react", "^3.34.0"): "^3.35.0",
    # community/playground/vitest-showcase
    ("community/playground/vitest-showcase/package.json", "devDependencies", "@cloudflare/vitest-pool-workers", "^0.12.21"): "^0.15.0",
    # playground/chakra-ui
    ("playground/chakra-ui/package.json", "dependencies", "@chakra-ui/react", "^3.34.0"): "^3.35.0",
    # playground/monorepo-top-level-deps/packages/project
    ("playground/monorepo-top-level-deps/packages/project/package.json", "devDependencies", "typescript", "5.9.3"): "6.0.3",
    ("playground/monorepo-top-level-deps/packages/project/package.json", "devDependencies", "vitest", "^4.0.18"): "^4.1.5",
}

def update_pkg(path: Path, changes: dict):
    text = path.read_text()
    data = json.loads(text)
    modified = False
    for (dep_type, pkg, old_val), new_val in changes.items():
        if dep_type in data and pkg in data[dep_type]:
            if data[dep_type][pkg] == old_val:
                data[dep_type][pkg] = new_val
                modified = True
    if modified:
        # Preserve 2-space indent
        path.write_text(json.dumps(data, indent=2) + "\n")
    return modified

def main():
    # Apply common bumps to ALL package.json files
    count = 0
    for pattern in ["*/package.json", "*/*/package.json", "*/*/*/package.json", "*/*/*/*/package.json"]:
        for p in REPO.glob(pattern):
            if update_pkg(p, COMMON_BUMPS):
                count += 1
                print(f"Updated common: {p.relative_to(REPO)}")

    # Apply specific bumps
    for (rel_path, dep_type, pkg, old_val), new_val in SPECIFIC_BUMPS.items():
        p = REPO / rel_path
        if not p.exists():
            print(f"MISSING: {rel_path}")
            continue
        changes = {(dep_type, pkg, old_val): new_val}
        if update_pkg(p, changes):
            count += 1
            print(f"Updated specific: {rel_path}")

    print(f"\nTotal files modified: {count}")

if __name__ == "__main__":
    main()

File: scripts/bump-packages.mjs
import { readFileSync, writeFileSync } from "fs";
import { execSync } from "child_process";
import { join, relative } from "path";

const REPO = "/home/vscode/repo";

const COMMON_BUMPS = [
  { depType: "devDependencies", pkg: "@cloudflare/vite-plugin", old: "1.30.1", new: "1.33.2" },
  { depType: "devDependencies", pkg: "@cloudflare/vite-plugin", old: "1.31.0", new: "1.33.2" },
  { depType: "devDependencies", pkg: "@cloudflare/workers-types", old: "4.20260331.1", new: "4.20260426.1" },
  { depType: "devDependencies", pkg: "wrangler", old: "4.77.0", new: "4.85.0" },
  { depType: "devDependencies", pkg: "wrangler", old: "4.79.0", new: "4.85.0" },
  { depType: "devDependencies", pkg: "@types/node", old: "24.10.4", new: "~25.6.0" },
  { depType: "devDependencies", pkg: "@types/node", old: "~24.12.0", new: "~25.6.0" },
  { depType: "devDependencies", pkg: "typescript", old: "5.9.3", new: "6.0.3" },
  { depType: "devDependencies", pkg: "typescript", old: "6.0.2", new: "6.0.3" },
  { depType: "dependencies", pkg: "@cloudflare/workers-types", old: "4.20260331.1", new: "4.20260426.1" },
  { depType: "dependencies", pkg: "@cloudflare/workers-types", old: "4.20260405.1", new: "4.20260426.1" },
  { depType: "dependencies", pkg: "@types/node", old: "24.10.4", new: "~25.6.0" },
  { depType: "dependencies", pkg: "@types/node", old: "~24.12.0", new: "~25.6.0" },
  { depType: "dependencies", pkg: "typescript", old: "5.9.3", new: "6.0.3" },
  { depType: "dependencies", pkg: "typescript", old: "6.0.2", new: "6.0.3" },
];

const SPECIFIC_BUMPS = [
  // docs
  { file: "docs/package.json", depType: "dependencies", pkg: "@base-ui/react", old: "^1.2.0", new: "^1.4.1" },
  { file: "docs/package.json", depType: "dependencies", pkg: "fumadocs-core", old: "^16.7.6", new: "^16.8.5" },
  { file: "docs/package.json", depType: "dependencies", pkg: "fumadocs-mdx", old: "^14.2.7", new: "^14.3.2" },
  { file: "docs/package.json", depType: "dependencies", pkg: "tailwindcss", old: "^4.1.13", new: "^4.2.4" },
  { file: "docs/package.json", depType: "devDependencies", pkg: "@fumadocs/base-ui", old: "^16.7.4", new: "^16.8.5" },
  { file: "docs/package.json", depType: "devDependencies", pkg: "@tailwindcss/vite", old: "^4.1.6", new: "^4.2.4" },
  { file: "docs/package.json", depType: "devDependencies", pkg: "oxfmt", old: "^0.43.0", new: "^0.47.0" },
  { file: "docs/package.json", depType: "devDependencies", pkg: "oxlint", old: "^1.58.0", new: "^1.62.0" },
  // playground/mantine
  { file: "playground/mantine/package.json", depType: "dependencies", pkg: "@mantine/core", old: "^9.0.0", new: "^9.1.1" },
  { file: "playground/mantine/package.json", depType: "dependencies", pkg: "@mantine/hooks", old: "^9.0.0", new: "^9.1.1" },
  { file: "playground/mantine/package.json", depType: "devDependencies", pkg: "postcss", old: "^8.5.8", new: "^8.5.12" },
  // playground/shadcn
  { file: "playground/shadcn/package.json", depType: "dependencies", pkg: "lucide-react", old: "^1.7.0", new: "^1.11.0" },
  { file: "playground/shadcn/package.json", depType: "dependencies", pkg: "react-hook-form", old: "^7.72.0", new: "^7.74.0" },
  { file: "playground/shadcn/package.json", depType: "dependencies", pkg: "react-resizable-panels", old: "^4.7.2", new: "^4.10.0" },
  { file: "playground/shadcn/package.json", depType: "dependencies", pkg: "tailwindcss", old: "^4.2.1", new: "^4.2.4" },
  { file: "playground/shadcn/package.json", depType: "devDependencies", pkg: "@tailwindcss/vite", old: "^4.2.1", new: "^4.2.4" },
  { file: "playground/shadcn/package.json", depType: "devDependencies", pkg: "shadcn", old: "^4.1.2", new: "^4.5.0" },
  // playground/resend
  { file: "playground/resend/package.json", depType: "dependencies", pkg: "@react-email/components", old: "^1.0.8", new: "^1.0.12" },
  { file: "playground/resend/package.json", depType: "dependencies", pkg: "@react-email/render", old: "^2.0.4", new: "^2.0.7" },
  { file: "playground/resend/package.json", depType: "dependencies", pkg: "resend", old: "^6.10.0", new: "^6.12.2" },
  // playground/storybook
  { file: "playground/storybook/package.json", depType: "devDependencies", pkg: "@storybook/addon-a11y", old: "^10.3.3", new: "^10.3.5" },
  { file: "playground/storybook/package.json", depType: "devDependencies", pkg: "@storybook/addon-docs", old: "^10.3.3", new: "^10.3.5" },
  { file: "playground/storybook/package.json", depType: "devDependencies", pkg: "@storybook/addon-links", old: "^10.3.3", new: "^10.3.5" },
  { file: "playground/storybook/package.json", depType: "devDependencies", pkg: "@storybook/addon-vitest", old: "^10.3.3", new: "^10.3.5" },
  { file: "playground/storybook/package.json", depType: "devDependencies", pkg: "@storybook/react", old: "^10.3.3", new: "^10.3.5" },
  { file: "playground/storybook/package.json", depType: "devDependencies", pkg: "@storybook/react-vite", old: "^10.3.3", new: "^10.3.5" },
  { file: "playground/storybook/package.json", depType: "devDependencies", pkg: "storybook", old: "^10.3.3", new: "^10.3.5" },
  // playground/content-collections
  { file: "playground/content-collections/package.json", depType: "dependencies", pkg: "@content-collections/core", old: "^0.14.3", new: "^0.15.0" },
  { file: "playground/content-collections/package.json", depType: "dependencies", pkg: "@content-collections/vite", old: "^0.2.9", new: "^0.3.0" },
  // playground/community/todo-serverquery-and-actions
  { file: "playground/community/todo-serverquery-and-actions/package.json", depType: "dependencies", pkg: "tailwindcss", old: "^4.2.1", new: "^4.2.4" },
  { file: "playground/community/todo-serverquery-and-actions/package.json", depType: "devDependencies", pkg: "@tailwindcss/vite", old: "^4.2.1", new: "^4.2.4" },
  // playground/database-do
  { file: "playground/database-do/package.json", depType: "dependencies", pkg: "kysely", old: "^0.28.15", new: "^0.28.16" },
  // community
  { file: "community/package.json", depType: "devDependencies", pkg: "@cloudflare/workers-types", old: "~4.20260331.1", new: "~4.20260426.1" },
  { file: "community/package.json", depType: "devDependencies", pkg: "vitest", old: "~4.1.2", new: "~4.1.5" },
  // community/playground/ark-ui-showcase
  { file: "community/playground/ark-ui-showcase/package.json", depType: "dependencies", pkg: "@ark-ui/react", old: "^5.35.0", new: "^5.36.2" },
  { file: "community/playground/ark-ui-showcase/package.json", depType: "dependencies", pkg: "lucide-react", old: "^1.7.0", new: "^1.11.0" },
  // community/playground/chakra-ui-showcase
  { file: "community/playground/chakra-ui-showcase/package.json", depType: "dependencies", pkg: "@chakra-ui/react", old: "^3.34.0", new: "^3.35.0" },
  // community/playground/vitest-showcase
  { file: "community/playground/vitest-showcase/package.json", depType: "devDependencies", pkg: "@cloudflare/vitest-pool-workers", old: "^0.12.21", new: "^0.15.0" },
  // playground/chakra-ui
  { file: "playground/chakra-ui/package.json", depType: "dependencies", pkg: "@chakra-ui/react", old: "^3.34.0", new: "^3.35.0" },
  // playground/monorepo-top-level-deps/packages/project
  { file: "playground/monorepo-top-level-deps/packages/project/package.json", depType: "devDependencies", pkg: "typescript", old: "5.9.3", new: "6.0.3" },
  { file: "playground/monorepo-top-level-deps/packages/project/package.json", depType: "devDependencies", pkg: "vitest", old: "^4.0.18", new: "^4.1.5" },
];

function updateJson(filePath, bumps) {
  const text = readFileSync(filePath, "utf-8");
  const data = JSON.parse(text);
  let modified = false;
  for (const bump of bumps) {
    const section = data[bump.depType];
    if (section && section[bump.pkg] === bump.old) {
      section[bump.pkg] = bump.new;
      modified = true;
    }
  }
  if (modified) {
    writeFileSync(filePath, JSON.stringify(data, null, 2) + "\n");
    console.log("Updated:", relative(REPO, filePath));
  }
  return modified;
}

let count = 0;

// Apply common bumps to all package.json files
const files = execSync("find /home/vscode/repo -name 'package.json' -not -path '*/node_modules/*' -not -path '*/.git/*'", { encoding: "utf-8" })
  .trim()
  .split("\n")
  .filter(Boolean);

for (const file of files) {
  if (updateJson(file, COMMON_BUMPS)) count++;
}

// Apply specific bumps
for (const bump of SPECIFIC_BUMPS) {
  const file = join(REPO, bump.file);
  if (updateJson(file, [{ depType: bump.depType, pkg: bump.pkg, old: bump.old, new: bump.new }])) {
    count++;
  }
}

console.log("\nTotal files modified:", count);
